File: CS/MvpvmNavigation/Services/ModuleActivator.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Reflection;
using System.Text;
using System.Threading.Tasks;

namespace MvpvmNavigation.Services {
    public interface IModuleActivator {
        object CreateModule(string moduleTypeName);
    }
    //
    class ModuleActivator : IModuleActivator {
        Assembly moduleAssembly;
        string rootNamespace;
        public ModuleActivator(Assembly moduleAssembly, string rootNamespace) {
            this.moduleAssembly = moduleAssembly;
            this.rootNamespace = rootNamespace;
        }
        public object CreateModule(string moduleTypeName) {
            return Activator.CreateInstance(moduleAssembly.GetType(rootNamespace + '.' + moduleTypeName));
        }
    }
}

File: VB/MvpvmNavigation/Services/ModuleActivator.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Reflection
Imports System.Text
Imports System.Threading.Tasks

Namespace MvpvmNavigation.Services
    Public Interface IModuleActivator
        Function CreateModule(ByVal moduleTypeName As String) As Object
    End Interface
    '
    Friend Class ModuleActivator
        Implements IModuleActivator

        Private moduleAssembly As System.Reflection.Assembly
        Private rootNamespace As String
        Public Sub New(ByVal moduleAssembly As System.Reflection.Assembly, ByVal rootNamespace As String)
            Me.moduleAssembly = moduleAssembly
            Me.rootNamespace = rootNamespace
        End Sub
        Public Function CreateModule(ByVal moduleTypeName As String) As Object Implements IModuleActivator.CreateModule
            Return Activator.CreateInstance(moduleAssembly.GetType(rootNamespace & "."c & moduleTypeName))
        End Function
    End Class
End Namespace
